### ISSUE TITLE
Getting a fully prefixed path from `ly_path_data2schema`

### ISSUE THREAD

**Issue by syyyr** [CESNET/libyang#921]
Hello,
I'm trying to use `libyang::Context::find_path` to access a node inside a choice statement:
```
#include <libyang/Libyang.hpp>

auto schema = R"(
module example-schema {
    prefix aha;
    namespace "http://example.com";

    container myCont {
        choice myChoice {
            case case1 {
                leaf leaf1 {
                    type string;
                }
            }
        }
    }
}
)";

int main(int argc, char* argv[])
{
    auto context = std::make_shared<libyang::Context>();
    context->parse_module_mem(schema, LYS_IN_YANG);

    auto leaf = context->find_path("/example-schema:myCont/example-schema:leaf1");
    return 0;
}
```
This doesn't work, because `libyang::Context::find_path` only finds this node with a path, which includes the choice and case node:
```
/example-schema:myCont/example-schema:myChoice/example-schema:case1/example-schema:leaf1
```
I found that the `ly_path_data2schema` function can convert the path to include these nodes Its output is:
```
/example-schema:myCont/myChoice/case1/leaf1
```
but now the path is not fully prefixed anymore and can't be used with `libyang::Context::find_path`. Is there a way to make the path fully prefixed again?

Thanks.

**Comment by fredgan** [comment]
You can use as this:
···cpp
#include <libyang/Libyang.hpp>
#include <libyang/Tree_Schema.hpp>

auto schema = R"(
module example-schema {
    prefix aha;
    namespace "http://example.com";

    container myCont {
        choice myChoice {
            case case1 {
                leaf leaf1 {
                    type string;
                }
            }
        }
    }
}
)";

int main(int argc, char* argv[])
{
    auto context = std::make_shared<libyang::Context>();
    auto module = context->parse_module_mem(schema, LYS_IN_YANG);
    auto schema_node = module->data();

    auto leaf = schema_node->find_path("/example-schema:myCont/myChoice/case1/leaf1");
    return 0;
}
···

**Comment by syyyr** [comment]
@fredgan
Hi, I need this to work for any top level module. In my program, I load various schemas in different ways, so I can't just save the module handle. I could parse the top level module from the string and then get the module handle and then I could do what you suggest, but parsing the string seems unnecessary (since libyang can already parse paths).

**Comment by rkrejci** [comment]
Hi, what about `libyang::Context::get_node`, would it work for you? The path is the data path (so `"/example-schema:myCont/example-schema:leaf1"` from your example should work) and `start` shouldn't be needed with the absolute paths.

**Comment by rkrejci** [comment]
Actually in the C++ wrapper you have to set `start` parameter, but in case the path is absolute, it will not be used for anything, it just must not be `nullptr`. I'm going to fix it in the bindings.

**Comment by rkrejci** [comment]
The problem with the `nullptr` should be fixed in 1.0.83, please check if the function does what you need.

**Comment by syyyr** [comment]
`libyang::Context::get_node` does what I want, thanks! But, there was a mistake in the last commit, so I fixed it in #925.

**Comment by rkrejci** [comment]
Thanks!
